### PR TITLE
Add capture* macros and improve suppress* ones.

### PR DESCRIPTION
**Pull request by ssfrr**
I think if you don't close the streams then the reader task will never finish. I also removed some assignments that didn't seem to be used anywhere.

This PR includes the commits for #2, which makes the diff messier but I'm not sure how to better handle dependent PRs. If you want this but not that, let me know and I'll rebase just these changes on top of master.

### CODE REVIEW

**Comment by ssfrr** [comment]
cc @vtjnash - are these doing the right thing now?

**Comment by Ismael-VC** [comment]
@ssfrr thank you very much for adding this functionality,  I would like to add all your changes, should I wait for @vtjnash to comment on this before merging?

**Comment by vtjnash** [comment]
Yes, this looks correct

File: test/runtests.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ end
 output = @capture_err begin
     warn("should get captured, not printed")
 end
-@test output == "WARNING: should get captured, not printed\n"
+@test output == "\e[1m\e[31mWARNING: should get captured, not printed\e[0m\n"
```
Comment (tkelman) [inline test/runtests.jl:12]: this should probably actually check `Base.have_color` so it can pass in either setting

**Pull request merged**